components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/main/java/org/wso2/carbon/identity/secret/mgt/core/IdentityProviderSecretsProcessorImpl.java

```diff
@@ -0,0 +1,133 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.secret.mgt.core;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.application.common.model.FederatedAuthenticatorConfig;
+import org.wso2.carbon.identity.application.common.model.IdentityProvider;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.secret.mgt.core.exception.SecretManagementException;
+import org.wso2.carbon.identity.secret.mgt.core.model.ResolvedSecret;
+import org.wso2.carbon.identity.secret.mgt.core.model.Secret;
+import org.wso2.carbon.identity.secret.mgt.core.model.SecretType;
+import static org.wso2.carbon.identity.secret.mgt.core.constant.SecretConstants.IDN_SECRET_TYPE_IDP_SECRETS;
+
+/**
+ * Identity provider secrets processor service implementation.
+ */
+public class IdentityProviderSecretsProcessorImpl implements IdentityProviderSecretsProcessor {
+
+    private final SecretManager secretManager;
+    private final SecretResolveManager secretResolveManager;
+
+    public IdentityProviderSecretsProcessorImpl() {
+
+        this.secretManager = new SecretManagerImpl();
+        this.secretResolveManager = new SecretResolveManagerImpl();
+    }
+
+    @Override
+    public IdentityProvider addOrUpdateIdPSecrets(IdentityProvider identityProvider) throws SecretManagementException {
+
+        for (FederatedAuthenticatorConfig fedAuthConfig : identityProvider.getFederatedAuthenticatorConfigs()) {
+            for (Property prop : fedAuthConfig.getProperties()) {
+                if (prop.isConfidential()) {
+                    String secretName = buildSecretName(identityProvider.getId(), fedAuthConfig.getName(), prop.getName());
+                    if (secretManager.isSecretExist(IDN_SECRET_TYPE_IDP_SECRETS, secretName)) {
+                        // Update existing secret property.
+                        updateExistingSecretProperty(secretName, prop);
+                        prop.setValue(buildSecretReference(secretName));
+                    } else {
+                        // Add secret to the DB.
+                        if (!StringUtils.isEmpty(prop.getValue())) {
+                            addNewIdpSecretProperty(secretName, prop);
+                            prop.setValue(buildSecretReference(secretName));
+                        }
+                    }
+                }
+            }
+        }
+
+        return identityProvider;
+    }
+
+    @Override
+    public IdentityProvider getIdPSecrets(IdentityProvider identityProvider)
```

Comment:
@sachiniWettasinghe @indeewari @kayathiri4 
This method signature looks odd. We can not have gettters mutating an object.
This method name should be changed to 
```
public IdentityProvider associateSecrets(IdentityProvider identityProvider)
```

Reply:
Resolved with [4cca0ec](https://github.com/wso2/carbon-identity-framework/pull/4430/commits/4cca0ec26dfa11c945d145f1a0ba1ad3d2cabd20)